ntoskrnl/io/pnpmgr/pnproot.c

```diff
@@ -461,15 +461,20 @@ EnumerateDevices(
     DeviceExtension = (PPNPROOT_FDO_DEVICE_EXTENSION)DeviceObject->DeviceExtension;
     KeAcquireGuardedMutex(&DeviceExtension->DeviceListLock);
 
-    BufferSize = sizeof(KEY_BASIC_INFORMATION) + (MAX_PATH + 1) * sizeof(WCHAR);
-    KeyInfo = ExAllocatePoolWithTag(PagedPool, BufferSize, TAG_PNP_ROOT);
+    KeyInfoSize = FIELD_OFFSET(KEY_BASIC_INFORMATION, Name) + 64 * sizeof(WCHAR);
```

Comment:
What's that magic value? It doesn't look that arbitrary?

Reply:
The point of "handle arbitrary subkey name length" is that now it really is arbitrary. It will work with a value of `1` (this is how I tested the change). `64` is a guess that should cover most key names. Do you think there's a better value?

Reply:
OK, thanks for the explanation :-)